connectors/sources/salesforce.py

```diff
@@ -79,32 +84,57 @@ async def get_token(self, enable_refresh=True):
         interval=RETRY_INTERVAL,
     )
     async def get_accounts(self):
-        # TODO prepare cache
+        if not await self._is_queryable("Account"):
+            return
 
-        query_builder = SalesforceSoqlBuilder("Account")
-        query_builder.with_id()
-        query_builder.with_default_metafields()
-        query_builder.with_fields(
-            self._select_queryable_fields(
-                [
-                    "Name",
-                    "Description",
-                    "BillingAddress",
-                    "Type",
-                    "Website",
-                    "Rating",
-                    "Department",
-                ]
-            )
-        )
-        query = query_builder.build()
+        # TODO handle pagination
+        query = await self._accounts_query()
+        resp = await self._yield_non_bulk_query_pages(query)
+        resp_json = await resp.json()
+        for record in resp_json.get("records", []):
+            yield self.doc_builder.build_account(record)
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+    )
+    async def get_opportunities(self):
+        if not await self._is_queryable("Opportunity"):
+            return
 
         # TODO handle pagination
+        query = await self._opportunities_query()
         resp = await self._yield_non_bulk_query_pages(query)
         resp_json = await resp.json()
         for record in resp_json.get("records", []):
-            record["_id"] = record["Id"]
-            yield record
+            yield self.doc_builder.build_oppoortunity(record)
+
+    async def prepare_caches(self):
+        await self._cache_queryable_sobjects()
+
+    async def _cache_queryable_sobjects(self):
+        resp = await self._api_request(
+            f"{self.base_url}{DESCRIBE_ENDPOINT}", "get", headers=self._auth_headers()
+        )
+        resp_json = await resp.json()
+        self.queryable_sobjects = [
+            x["name"].lower() for x in resp_json["sobjects"] if x["queryable"] is True
+        ]
+
+    async def _is_queryable(self, sobject):
+        return sobject.lower() in self.queryable_sobjects
+
+    async def _select_queryable_fields(self, sobject, fields):
+        """User settings can cause fields to be non-queryable
+        This causes
+        """
+        endpoint = DESCRIBE_SOBJECT_ENDPOINT.replace("<sobject>", sobject)
+        resp = await self._api_request(
+            f"{self.base_url}{endpoint}", "get", headers=self._auth_headers()
+        )
+        resp_json = await resp.json()
+        queryable_fields = [f["name"] for f in resp_json["fields"]]
+        return [f for f in fields if f in queryable_fields]
```

Comment:
Intuitively, it can be cached too. Would it be a good idea to unite this method with `_cache_queryable_objects` as they kinda prepare the class for working?

Reply:
Same as above: 
I can do this if I introduce [asyncstdlib](https://asyncstdlib.readthedocs.io/en/stable/index.html#) or [async-property](https://pypi.org/project/async-property/).

Reply:
This is now cached 💾